src/Altinn.App.Core/Models/Layout/BaseComponent.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Immutable;
+using System.Text.Json;
+
+using Altinn.App.Core.Models.Expressions;
+
+namespace Altinn.App.Core.Models.Layout;
+
+
+/// <summary>
+/// Inteface to be able to handle all most components same way.
+/// </summary>
+/// <remarks>
+/// See <see cref="GroupComponent" /> for any components that handle children.
+/// Includes <see cref="DataModelBindings" /> that will be initialized to an empty dictionary
+/// for components that don't have them.
+/// </remarks>
+public class BaseComponent
+{
+    /// <summary>
+    /// Constructor that ensures n
+    /// </summary>
+    public BaseComponent(string id, string type, IReadOnlyDictionary<string, string>? dataModelBindings, Expression? hidden, Expression? required, Expression? readOnly, IReadOnlyDictionary<string, string>? extra)
+    {
+        Id = id;
```

Comment:
Should we validate Id and Type in the constructor and throw an InvalidArgumentException if they are null or empty?

Reply:
Ok, but that's another class and only compile time checks. Isn't it better if the class that actually has the properties ensures they are correct and that it leaves the constructor with the object in valid state?